### ISSUE TITLE
fails to test without feature "quickcheck"

### ISSUE THREAD

**Issue by jonassmedegaard** [rust-crdt/rust-crdt#136]
This fails: `cargo test --no-default-features`

**Comment by davidrusu** [comment]
Thanks for the report, fixed in #137

**Comment by jonassmedegaard** [comment]
Sorry, but release v7.3.1 still fails for me, like this:
```
   Compiling crdts v7.3.1 (/usr/share/cargo/registry/crdts-7.3.1)
     Running `CARGO=/usr/bin/cargo CARGO_CRATE_NAME=crdts CARGO_MANIFEST_DIR=/usr/share/cargo/registry/crdts-7.3.1 CARGO_PKG_AUTHORS='Tyler Neely <t@jujit.su>:David Rusu <davidrusu.me@gmail.com>' CARGO_PKG_DESCRIPTION='Practical, serializable, thoroughly tested CRDTs' CARGO_PKG_HOMEPAGE='https://github.com/rust-crdt/rust-crdt' CARGO_PKG_LICENSE=Apache-2.0 CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=crdts CARGO_PKG_REPOSITORY='https://github.com/rust-crdt/rust-crdt' CARGO_PKG_RUST_VERSION='' CARGO_PKG_VERSION=7.3.1 CARGO_PKG_VERSION_MAJOR=7 CARGO_PKG_VERSION_MINOR=3 CARGO_PKG_VERSION_PATCH=1 CARGO_PKG_VERSION_PRE='' CARGO_PRIMARY_PACKAGE=1 LD_LIBRARY_PATH='/tmp/tmp.38yk9eJFGX/target/debug/deps:/usr/lib:/usr/lib/libeatmydata' /usr/bin/sccache rustc --crate-name crdts --edition=2021 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="merkle"' -C metadata=9a13b28aea869f41 -C extra-filename=-9a13b28aea869f41 --out-dir /tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps --target x86_64-unknown-linux-gnu -C incremental=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/incremental -L dependency=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps -L dependency=/tmp/tmp.38yk9eJFGX/target/debug/deps --extern serde=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps/libserde-572aa589169f87e1.rmeta --extern tiny_keccak=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps/libtiny_keccak-52500e2ab4e0d6d7.rmeta -C debuginfo=2 --cap-lints warn -C linker=x86_64-linux-gnu-gcc -C link-arg=-Wl,-z,relro --remap-path-prefix /usr/share/cargo/registry/crdts-7.3.1=/usr/share/cargo/registry/crdts-7.3.1 --remap-path-prefix /tmp/tmp.38yk9eJFGX/registry=/usr/share/cargo/registry`
warning: `derive_more` (lib) generated 11 warnings
     Running `CARGO=/usr/bin/cargo CARGO_CRATE_NAME=reset_remove CARGO_MANIFEST_DIR=/usr/share/cargo/registry/crdts-7.3.1 CARGO_PKG_AUTHORS='Tyler Neely <t@jujit.su>:David Rusu <davidrusu.me@gmail.com>' CARGO_PKG_DESCRIPTION='Practical, serializable, thoroughly tested CRDTs' CARGO_PKG_HOMEPAGE='https://github.com/rust-crdt/rust-crdt' CARGO_PKG_LICENSE=Apache-2.0 CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=crdts CARGO_PKG_REPOSITORY='https://github.com/rust-crdt/rust-crdt' CARGO_PKG_RUST_VERSION='' CARGO_PKG_VERSION=7.3.1 CARGO_PKG_VERSION_MAJOR=7 CARGO_PKG_VERSION_MINOR=3 CARGO_PKG_VERSION_PATCH=1 CARGO_PKG_VERSION_PRE='' CARGO_PRIMARY_PACKAGE=1 LD_LIBRARY_PATH='/tmp/tmp.38yk9eJFGX/target/debug/deps:/usr/lib:/usr/lib/libeatmydata' /usr/bin/sccache rustc --crate-name reset_remove --edition=2021 examples/reset_remove.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --test --cfg 'feature="merkle"' -C metadata=4955b7658edfd38d -C extra-filename=-4955b7658edfd38d --out-dir /tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/examples --target x86_64-unknown-linux-gnu -C incremental=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/incremental -L dependency=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps -L dependency=/tmp/tmp.38yk9eJFGX/target/debug/deps --extern crdts=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps/libcrdts-9a13b28aea869f41.rlib --extern derive_more=/tmp/tmp.38yk9eJFGX/target/debug/deps/libderive_more-0e336671c8195696.so --extern quickcheck_macros=/tmp/tmp.38yk9eJFGX/target/debug/deps/libquickcheck_macros-10c1d1f938224492.so --extern rand=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps/librand-262ba871026b1e5d.rlib --extern serde=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps/libserde-572aa589169f87e1.rlib --extern tiny_keccak=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps/libtiny_keccak-52500e2ab4e0d6d7.rlib -C debuginfo=2 --cap-lints warn -C linker=x86_64-linux-gnu-gcc -C link-arg=-Wl,-z,relro --remap-path-prefix /usr/share/cargo/registry/crdts-7.3.1=/usr/share/cargo/registry/crdts-7.3.1 --remap-path-prefix /tmp/tmp.38yk9eJFGX/registry=/usr/share/cargo/registry`
     Running `CARGO=/usr/bin/cargo CARGO_CRATE_NAME=test CARGO_MANIFEST_DIR=/usr/share/cargo/registry/crdts-7.3.1 CARGO_PKG_AUTHORS='Tyler Neely <t@jujit.su>:David Rusu <davidrusu.me@gmail.com>' CARGO_PKG_DESCRIPTION='Practical, serializable, thoroughly tested CRDTs' CARGO_PKG_HOMEPAGE='https://github.com/rust-crdt/rust-crdt' CARGO_PKG_LICENSE=Apache-2.0 CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=crdts CARGO_PKG_REPOSITORY='https://github.com/rust-crdt/rust-crdt' CARGO_PKG_RUST_VERSION='' CARGO_PKG_VERSION=7.3.1 CARGO_PKG_VERSION_MAJOR=7 CARGO_PKG_VERSION_MINOR=3 CARGO_PKG_VERSION_PATCH=1 CARGO_PKG_VERSION_PRE='' CARGO_PRIMARY_PACKAGE=1 CARGO_TARGET_TMPDIR=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/tmp LD_LIBRARY_PATH='/tmp/tmp.38yk9eJFGX/target/debug/deps:/usr/lib:/usr/lib/libeatmydata' /usr/bin/sccache rustc --crate-name test --edition=2021 test/test.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --test --cfg 'feature="merkle"' -C metadata=9b0d3141ecd23c4a -C extra-filename=-9b0d3141ecd23c4a --out-dir /tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps --target x86_64-unknown-linux-gnu -C incremental=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/incremental -L dependency=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps -L dependency=/tmp/tmp.38yk9eJFGX/target/debug/deps --extern crdts=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps/libcrdts-9a13b28aea869f41.rlib --extern derive_more=/tmp/tmp.38yk9eJFGX/target/debug/deps/libderive_more-0e336671c8195696.so --extern quickcheck_macros=/tmp/tmp.38yk9eJFGX/target/debug/deps/libquickcheck_macros-10c1d1f938224492.so --extern rand=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps/librand-262ba871026b1e5d.rlib --extern serde=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps/libserde-572aa589169f87e1.rlib --extern tiny_keccak=/tmp/tmp.38yk9eJFGX/target/x86_64-unknown-linux-gnu/debug/deps/libtiny_keccak-52500e2ab4e0d6d7.rlib -C debuginfo=2 --cap-lints warn -C linker=x86_64-linux-gnu-gcc -C link-arg=-Wl,-z,relro --remap-path-prefix /usr/share/cargo/registry/crdts-7.3.1=/usr/share/cargo/registry/crdts-7.3.1 --remap-path-prefix /tmp/tmp.38yk9eJFGX/registry=/usr/share/cargo/registry`
error[E0463]: can't find crate for `quickcheck`
 --> test/test.rs:2:1
  |
2 | extern crate quickcheck;
  | ^^^^^^^^^^^^^^^^^^^^^^^^ can't find crate

error[E0432]: unresolved import `crdts::glist`
 --> test/glist.rs:3:12
  |
3 | use crdts::glist::{GList, Op};
  |            ^^^^^ could not find `glist` in `crdts`

error[E0432]: unresolved import `crdts::Identifier`
 --> test/glist.rs:4:27
  |
4 | use crdts::{CmRDT, CvRDT, Identifier};
  |                           ^^^^^^^^^^ no `Identifier` in the root

error[E0432]: unresolved import `crdts::list`
 --> test/list.rs:1:12
  |
1 | use crdts::list::{List, Op};
  |            ^^^^ could not find `list` in `crdts`

error[E0432]: unresolved import `num`
 --> test/glist.rs:1:5
  |
1 | use num::BigRational;
  |     ^^^ help: a similar path exists: `std::num`

error: cannot find macro `quickcheck` in this scope
   --> test/map.rs:578:1
    |
578 | quickcheck! {
    | ^^^^^^^^^^
    |
note: `quickcheck` is imported here, but it is an unresolved item, not a macro
   --> test/test.rs:2:1
    |
2   | extern crate quickcheck;
    | ^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot find macro `quickcheck` in this scope
   --> test/mvreg.rs:142:1
    |
142 | quickcheck! {
    | ^^^^^^^^^^
    |
note: `quickcheck` is imported here, but it is an unresolved item, not a macro
   --> test/test.rs:2:1
    |
2   | extern crate quickcheck;
    | ^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot find macro `quickcheck` in this scope
  --> test/orswot.rs:13:1
   |
13 | quickcheck! {
   | ^^^^^^^^^^
   |
note: `quickcheck` is imported here, but it is an unresolved item, not a macro
  --> test/test.rs:2:1
   |
2  | extern crate quickcheck;
   | ^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot find macro `quickcheck` in this scope
 --> test/vclock.rs:5:1
  |
5 | quickcheck! {
  | ^^^^^^^^^^
  |
note: `quickcheck` is imported here, but it is an unresolved item, not a macro
 --> test/test.rs:2:1
  |
2 | extern crate quickcheck;
  | ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `ResetRemove`
 --> test/map.rs:1:66
  |
1 | use crdts::{map, mvreg, CmRDT, CvRDT, Dot, DotRange, MVReg, Map, ResetRemove, VClock};
  |                                                                  ^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `ctx::ReadCtx`
 --> test/orswot.rs:4:13
  |
4 | use crdts::{ctx::ReadCtx, orswot::Op, *};
  |             ^^^^^^^^^^^^

warning: unused import: `std::cmp::Ordering`
 --> test/vclock.rs:3:5
  |
3 | use std::cmp::Ordering;
  |     ^^^^^^^^^^^^^^^^^^

Some errors have detailed explanations: E0432, E0463.
For more information about an error, try `rustc --explain E0432`.
warning: `crdts` (test "test") generated 3 warnings
error: could not compile `crdts` due to 9 previous errors; 3 warnings emitted
```

**Comment by davidrusu** [comment]
How are you building the crate?

**Comment by davidrusu** [comment]
Thanks for making the effort to package up the crate, and you are right, the crate should build under any combination of features so it is a bug if that's not the case.

I'll take a closer look at the build logs I found on the tracker: https://ci.debian.net/data/autopkgtest/unstable/amd64/r/rust-crdts/36473501/log.gz

**Comment by davidrusu** [comment]
Ok, I think I've resolved this now in #138 and I've published a new release `v7.3.2`
 
Please let me know if you still see build failures
